drivers/osdp/src/osdp_phy.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2020 Siddharth Chandrasekaran <siddharth@embedjournal.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(osdp, CONFIG_OSDP_LOG_LEVEL);
+
+#include <string.h>
+#include "osdp_common.h"
+
+#define TAG "PHY: "
+
+#define PKT_CONTROL_SQN			0x03
+#define PKT_CONTROL_CRC			0x04
+#define PKT_CONTROL_SCB			0x08
+
+struct osdp_packet_header {
+	uint8_t mark;
+	uint8_t som;
+	uint8_t pd_address;
+	uint8_t len_lsb;
+	uint8_t len_msb;
+	uint8_t control;
+	uint8_t data[];
+};
+
+uint8_t compute_checksum(uint8_t *msg, int length)
```

Comment:
You should probably have the functions static here and below. If you want to use them in other files, then please use proper namespace for them (osdp_phy_...)

Reply:
Done.